test/test-constraint-group.jl

```diff
@@ -0,0 +1,180 @@
+@testset "Test add_group_constraints!" begin
+    # Setup a temporary DuckDB connection and model
+    connection = DBInterface.connect(DuckDB.DB)
+    model = JuMP.Model()
+
+    # Create mock tables for testing using register_data_frame
+
+    # Asset table - defines which assets belong to which group
+    asset_rows = [
+        ("wind_1", "renewable_group", 100.0, 2025),
+        ("wind_2", "renewable_group", 150.0, 2025),
+        ("solar_1", "renewable_group", 80.0, 2025),
+        ("gas_1", "fossil_group", 200.0, 2025),
+        ("gas_2", "fossil_group", 250.0, 2025),
+    ]
+    asset = DataFrame(asset_rows, [:asset, :group, :capacity, :milestone_year])
+    DuckDB.register_data_frame(connection, asset, "asset")
+
+    # Group asset table - defines the groups
+    group_asset_rows = [("renewable_group", "renewable"), ("fossil_group", "fossil")]
+    group_asset = DataFrame(group_asset_rows, [:name, :type])
+    DuckDB.register_data_frame(connection, group_asset, "group_asset")
+
+    # Variable assets investment table
+    var_assets_investment_rows = [
+        (1, "wind_1", 2025, true, 100.0, 500.0),
+        (2, "wind_2", 2025, true, 150.0, 600.0),
+        (3, "solar_1", 2025, true, 80.0, 400.0),
+        (4, "gas_1", 2025, true, 200.0, 800.0),
+        (5, "gas_2", 2025, true, 250.0, 1000.0),
+    ]
+    var_assets_investment = DataFrame(
+        var_assets_investment_rows,
+        [:id, :asset, :milestone_year, :investment_integer, :capacity, :investment_limit],
+    )
+    DuckDB.register_data_frame(connection, var_assets_investment, "var_assets_investment")
+
+    # Group max investment limit constraints
+    cons_group_max_investment_limit_rows =
+        [(1, "renewable_group", 1000.0), (2, "fossil_group", 1200.0)]
+    cons_group_max_investment_limit =
+        DataFrame(cons_group_max_investment_limit_rows, [:id, :name, :max_investment_limit])
+    DuckDB.register_data_frame(
+        connection,
+        cons_group_max_investment_limit,
+        "cons_group_max_investment_limit",
+    )
+
+    # Group min investment limit constraints
+    cons_group_min_investment_limit_rows =
+        [(1, "renewable_group", 200.0), (2, "fossil_group", 100.0)]
+    cons_group_min_investment_limit =
+        DataFrame(cons_group_min_investment_limit_rows, [:id, :name, :min_investment_limit])
+    DuckDB.register_data_frame(
+        connection,
+        cons_group_min_investment_limit,
+        "cons_group_min_investment_limit",
+    )
+
+    # Create empty tables for other investment variables that are required
```

Comment:
Do I understand this correctly that these empty tables are created explicitly because the code being tested expects this, but it's created elsewhere?

Either way, I don't think this is the correct way to do it, because it is essentially an "alternate implementation" of wherever Tulipa creates the tables.  Unless I recall incorrectly, the tables are created from the "schema" files in JSON format.  That should be used here as well, otherwise this test will go out of sync when the schema changes.

**PS:** If there's no generalised function that does the create table from the JSON files, then maybe this is also the opportunity to do a "good samaritan refactor" and add that feature in this unrelated PR.